armory/data/datasets.py

```diff
@@ -290,7 +301,39 @@ def enum_index(i, x):
         out, _ = tf.raw_ops.ListDiff(x=i, y=index_tensor, out_idx=tf.int64)
         return tf.equal(tf.size(out), 0)
 
-    return dataset.enumerate().filter(enum_index).map(lambda i, x: x), indexed_size
+    return dataset.enumerate().filter(enum_index).map(lambda i, x: x), num_valid_indices
+
+
+def filter_by_class(dataset: "tf.data.Dataset", class_ids: Union[list, int]):
+    """
+    class_ids must be an int or list of ints
+
+    returns the dataset filtered by class id, keeping elements with label in class_ids
+    """
+    logger.info(f"Filtering dataset to the following class IDs: {class_ids}")
+    if len(class_ids) == 0:
+        raise ValueError(
+            "The specified dataset 'class_ids' param must have at least one value"
+        )
+
+    def _filter_by_class(x, y, classes_to_keep=tf.constant(class_ids, dtype=tf.int64)):
+        isallowed_array = tf.equal(classes_to_keep, tf.cast(y, tf.int64))
+        isallowed = tf.reduce_sum(tf.cast(isallowed_array, tf.int64))
+        return tf.greater(isallowed, tf.constant(0, dtype=tf.int64))
+
+    filtered_ds = dataset.filter(_filter_by_class)
+
+    if tf.executing_eagerly():
```

Comment:
Filtering by class is already slow since it has to iterate over the whole dataset when `dataset.filter(_filter_by_class) is called. This process below of computing the filtered dataset size slows things down even further by iterating over the whole (filtered) dataset. It's also a bit messy that the approach differs based on the value of `tf.executing_eagerly()`. 

Ideally there would be a way to keep track of the filtered dataset size during the filtering of the dataset. Leaving this PR as WIP until I'm more confident in whether this is doable

Reply:
removing the WIP since I can't find a way around this. 

For most datasets it's relatively quick to filter the test set to a single class, < 3 seconds. However, it's taking ~20sec for So2Sat and ~2min for UCF
So2Sat: ~20sec
UCF: ~140sec

Reply:
This seems pretty reasonable for now. It took me 15 sec on so2sat and 44 sec on ucf101.

However, I see a big problem if this is done on a train split. Should we put a warning?